repos/system_upgrade/el7toel8/libraries/rpms.py

```diff
@@ -16,3 +17,37 @@ def get_installed_rpms():
                                                                                               RC=err.exit_code)
         stdlib.api.current_logger().error(error)
         return []
+
+
+def create_lookup(model, field, key, context=stdlib.api):
+    """
+    Create a lookup set from one of the model fields.
+
+    :param model: model class
+    :param field: model field, its value will be taken for lookup data
+    :param key: property of the field's data that will be used to build a resulting set
+    :param context: context of the execution
+    """
+    data = getattr(next((m for m in context.consume(model)), model()), field)
+    try:
+        return {getattr(obj, key) for obj in data} if data else {}
+    except TypeError:
+        # data is not iterable, not lookup can be built
+        stdlib.api.current_logger().error(
```

Comment:
Shouldn't we use context here as well? i.e.
`stdlib.api.current_logger()` → `context.current_logger()` ?

Reply:
@shaded-enmity no not really because context can be an actor and that doesn’t have current_logger